jans-cedarling/cedarling/src/authz/entities/create.rs

```diff
@@ -50,20 +52,25 @@ impl<'a> EntityMetadata<'a> {
     pub fn create_entity(
         &'a self,
         schema: &'a CedarSchemaJson,
-        data: &'a TokenPayload,
+        token: &Token,
         parents: HashSet<EntityUid>,
         claim_mapping: &ClaimMappings,
-    ) -> Result<cedar_policy::Entity, CedarPolicyCreateTypeError> {
+    ) -> Result<cedar_policy::Entity, CreateCedarEntityError> {
         let entity_uid = build_entity_uid(
             self.entity_type.full_type_name().as_str(),
-            data.get_payload(self.entity_id_data_key)?.as_str()?,
+            token
+                .get_claim(self.entity_id_data_key)
+                .ok_or(CreateCedarEntityError::MissingClaim(
+                    self.entity_id_data_key.to_string(),
+                ))?
+                .as_str()?,
         )?;
 
         create_entity(
             entity_uid,
             &self.entity_type,
             schema,
-            data,
+            token.data(),
             parents,
             claim_mapping,
         )
```

Comment:
```suggestion
    pub fn create_entity(
        &'a self,
        schema: &'a CedarSchemaJson,
        token: &TokenData,
        parents: HashSet<EntityUid>,
        claim_mapping: &ClaimMappings,
    ) -> Result<cedar_policy::Entity, CreateCedarEntityError> {
        let entity_uid = build_entity_uid(
            self.entity_type.full_type_name().as_str(),
            token
                .get_claim(self.entity_id_data_key)
                .ok_or(CreateCedarEntityError::MissingClaim(
                    self.entity_id_data_key.to_string(),
                ))?
                .as_str()?,
        )?;

        create_entity(
            entity_uid,
            &self.entity_type,
            schema,
            token,
            parents,
            claim_mapping,
        )
```

Reply:
I tried to do that at first but the `get_record_expression` creates new tokens inside it for some reason instead of just creating a new struct.

Reply:
refactored in [18ed767](https://github.com/JanssenProject/jans/pull/10436/commits/18ed7673bc485b13e045d2159b04de12cfc8641a)